review and improve instructions on PR blocking for Code (#1820)
semgrep/semgrep-docs: docs/semgrep-ci/configuring-blocking-and-errors-in-ci.md

append_help_link: true
description: "Configure how Semgrep in CI pipelines handles errors and blocks findings."
tags:
 - Deployment
title: Configure blocking findings
hide_title: true
---

import BlockFindingsErrorsConfigs from "/src/components/reference/_block-findings-errors-configs.mdx"

> # Configure handling of blocking findings and errors

Reviewer: ```suggestion
# Handling blocking findings and errors
```
Feel free to ignore this suggestion because I am not sure it is any better, having an `-ing` verb and `-ing` adjective.

Author: I do think it is a bit clearer!

This article documents how Semgrep handles blocking findings and errors and how you can change Semgrep's default behavior.

## Blocking findings

Blocking findings are those identified by Semgrep Code using rules defined in Semgrep AppSec Platform's [Policies page](https://semgrep.dev/orgs/-/policies) and are set to **Block** mode. You can avoid blocking findings by removing rules or by switching the rule mode to **Monitor**, **Comment**, or **Disabled**.

> If you do **not** use Semgrep AppSec Platform with Semgrep in CI (that is, you are using a **stand-alone setup**), all Semgrep findings are blocking findings. The existence of any findings means that Semgrep returns an exit code of `1`, which you can use to block your PRs or MRs.

Reviewer: ```suggestion
If you do **not** use Semgrep AppSec Platform with Semgrep in CI (that is, you are using a **stand-alone setup**), all Semgrep findings are blocking findings. The existence of any findings means that Semgrep returns an exit code of `1`, which you can use to block your PRs or MRs.
```

Blocking findings are those identified by Semgrep Code using rules defined in Semgrep AppSec Platform's [Policies page](https://semgrep.dev/orgs/-/policies) and are set to **Block** mode. You can avoid blocking findings by removing rules or by switching the rule mode to **Monitor**, **Comment**, or **Disabled**.

Blocking findings can be defined as:
If you do **not** use Semgrep AppSec Platform with Semgrep in CI (that is, you are using a **stand-alone setup**), all Semgrep findings are blocking findings. The existence of any findings means that Semgrep returns an exit code of `1`, which you can use to block your PRs or MRs.

- Findings defined in the [Policies page](https://semgrep.dev/orgs/-/policies) of Semgrep AppSec Platform. Avoid blocking findings by removing rules from the **Block** rule mode of the [Policies page](https://semgrep.dev/orgs/-/policies).
- If you do **not** use Semgrep AppSec Platform with Semgrep in CI (that is, you are using a **stand-alone setup**), blocking findings encompass all Semgrep findings. Any finding in this setup blocks your PRs or MRs.
## Semgrep's default behavior regarding blocking findings and errors

> By default, Semgrep does not block your pipeline when it encounters an internal error. In case of an internal error, Semgrep sends an anonymous crash report to a crash-reporting server and does not block your CI pipeline. To change the default configuration, see the sections below.

Reviewer: I would suggest explaining internal errors and making sure the user is aware they are apart from blocking findings.

> By default, Semgrep does not block your pipeline when it encounters an internal error. In case of an internal error, Semgrep sends an anonymous crash report to a crash-reporting server and does not block your CI pipeline. To change the default configuration, see the sections below.

Author: Let me know what you think of the modified version of this information on line 29

> By default, Semgrep does not block your pipeline when it encounters an internal error. In case of an internal error, Semgrep sends an anonymous crash report to a crash-reporting server and does not block your CI pipeline. To change the default configuration, see the sections below.

Reviewer: Looks good!

> ## Configuration options for blocking findings and errors

Reviewer: ```suggestion
## Configuration options for blocking findings and errors in CI
```
Just to separate this from SMS, where I don't think the user can configure the command.

 
You can configure, change, or revert to the default setup of blocking findings and errors in your CI pipeline by passing one of the following options in the `semgrep.yml` file used to configure and run Semgrep in your CI pipeline:

| CI option | Description |
|------------------------------------------------|-------------------------------------|
| `semgrep ci` or `semgrep ci --suppress-errors` | Default. CI **fails** on blocking findings, but **passes** on internal errors. |
| `semgrep ci --no-suppress-errors` | CI **fails** on blocking findings and internal errors. |
| <code>semgrep ci &vert;&vert; true</code> | CI **passes** on blocking findings and internal errors. |

> To change Semgrep's behavior, modify the `semgrep.yml`, specifically the `semgrep ci` command to the CI option that best fits your needs. For example, GitHub users would include the following under the `run` key:

Reviewer: ```suggestion
To change Semgrep's behavior, modify your pipeline or job file, specifically the `semgrep ci` command, to the CI option that best fits your needs. For example, GitHub users should edit the `semgrep.yml` workflow file and include the following under the `run` key:
```
Not all workflow files are `semgrep.yml`, so I generalized the tidbit and specificed `semgrep.yml` for our github example.
